04.Keyframe/main.cpp

```diff
@@ -142,23 +142,28 @@ int main()
 			
 		asset::SPushConstantRange range[1] = {asset::ISpecializedShader::ESS_VERTEX,0u,sizeof(core::matrix4SIMD)};
 
-		auto createSpecializedShaderFromSource = [=](const char* source, asset::ISpecializedShader::E_SHADER_STAGE stage)
+		auto createSpecializedShaderFromSource = [=](const char* source, asset::ISpecializedShader::E_SHADER_STAGE stage, asset::IShaderCompiler::SCompilerOptions&& options)
 		{
-			// TODO: Update: should use the compiler set from asset manager
-			auto spirv = device->getAssetManager()->getGLSLCompiler()->createSPIRVFromGLSL(source, stage, "main", "runtimeID");
+			options.stage = stage;
+			options.entryPoint = "main";
+			options.preprocessorOptions.sourceIdentifier = "runtimeID";
+			auto spirv = device->getAssetManager()->getCompilerSet()->compileToSPIRV(source, options);
 			return core::make_smart_refctd_ptr<asset::ICPUSpecializedShader>(std::move(spirv),asset::ICPUSpecializedShader::SInfo{ nullptr,nullptr,"main",stage });
 		};
 		// origFilepath is only relevant when you have filesystem #includes in your shader
+		auto system = nbl::core::make_smart_refctd_ptr<nbl::system::CSystemWin32>();
 		auto createSpecializedShaderFromSourceWithIncludes = [&](const char* source, asset::ISpecializedShader::E_SHADER_STAGE stage, const char* origFilepath)
 		{
-			auto resolved_includes = device->getAssetManager()->getGLSLCompiler()->resolveIncludeDirectives(source, stage, origFilepath);
-			return createSpecializedShaderFromSource(reinterpret_cast<const char*>(resolved_includes->getContent()->getPointer()), stage);
+			asset::IShaderCompiler::SCompilerOptions options;
+			options.preprocessorOptions.includeFinder = new asset::IShaderCompiler::CIncludeFinder(system);
+			options.preprocessorOptions.includeFinder->addSearchPath(origFilepath, options.preprocessorOptions.includeFinder->getDefaultFileSystemLoader());
```

Comment:
Again, no need to addSearchPaths, I see why you felt like doing this, but it does nothing, the origFilepath are just names for our shader which go into `sourceIdentifier`

But this example seems way out of date so  I'll revert everything.